func.go

```diff
@@ -65,6 +65,23 @@ func RegisterLibFunc(fptr interface{}, handle uintptr, name string) {
 // There are some limitations when using RegisterFunc on Linux. First, there is no support for function arguments.
 // Second, float32 and float64 arguments and return values do not work when CGO_ENABLED=1. Otherwise, Linux
 // has the same feature parity as Darwin.
+//
+// # Memory
+//
+// In general it is not possible for purego to guarantee the lifetimes of objects returned or received from
+// calling functions using RegisterFunc. For arguments to a C function is it important that the C function doesn't
+// hold onto a reference to Go memory. This is the same as the [Cgo rules]: https://pkg.go.dev/cmd/cgo#hdr-Go_references_to_C
+//
+// However, there are some special cases. When passing a string as an argument if the string does not end in a null
+// terminated byte (\x00) then the string will be copied into Go memory maintained by purego. The memory is only valid for
+// that specific call. Therefore, if the C code keeps a reference to that string it may become invalid at some
+// undefined time. However, if the string does already contain a null-terminated byte then no copy is done.
+// It is then the responsibility of the caller to ensure the string stays alive as long as it's needed in C memory.
+// This can be done using runtime.KeepAlive. When a C function returns a null-terminated pointer to char a Go string
+// can be used. Purego will allocate a new string in Go memory and copy the data over. This string will
+// be garbage collected whenever Go decides it's no longer referenced. If the pointer to char is not null-terminated or
+// must continue to point to C memory (because it's a buffer for example) then use a pointer to byte and then convert
+// that to a slice using unsafe.Slice.
```

Comment:
So, if we want to manage the lifetime of the returning value from C, we should not use this automatic conversion, right? Let's add this caveats.